collect_app/src/main/java/org/odk/collect/android/geo/MapboxMapFragment.java

```diff
@@ -356,12 +355,29 @@ private void resetLocationComponent() {
         }
     }
 
-    @Override public int addMarker(MapPoint point, boolean draggable, @IconAnchor String iconAnchor) {
+    @Override public int addMarker(MapPoint point, boolean draggable, @IconAnchor String iconAnchor, int iconDrawableId) {
         int featureId = nextFeatureId++;
-        features.put(featureId, new MarkerFeature(featureId, symbolManager, point, draggable, iconAnchor));
+        features.put(featureId, new MarkerFeature(featureId, symbolManager, point, draggable, iconAnchor, iconDrawableId));
         return featureId;
     }
 
+    @Override public void displayMarkers() {
+        List<SymbolOptions> symbolOptions = new ArrayList<>();
+
+        for (Map.Entry<Integer, MapFeature> entry : features.entrySet()) {
```

Comment:
This is pretty nasty as I said during the meeting... For example I had to change features to `LinkedHashMap` to make sure the order is preserved.
I don't love it but with current architecture thi is the easiest solution. We might want to rework it in the future.

Reply:
I agree this doesn't feel ideal. Did you consider adding a `addMarkers` method to the interface that takes in a `List<MapPoint>` and returns a `List<Integer>`? In the Google case, it would call `addMarker` in a loop and in the other two it would use the more performant add all functionality.

Reply:
Yes. Maybe it would be a bit better it depends...
First of all passing a list of `MapPoint` would not be enough. I need at least icons for every point so that would need to be two list or a map which would be a ix of both.
Another problem is that I would end up with exactly the same nasty iterating through map of features to build the list of `symbolOptions` and then setup symbols in every `markerFeature` after displaying them.

Reply:
This rather requires more changes in the way we create those features.